lib/shelley/test/unit/Cardano/Wallet/Shelley/TransactionSpec.hs

```diff
@@ -2227,6 +2232,50 @@ balanceTransactionSpec = do
             it "may fail"
                 $ property prop_balanceTransactionUnresolvedInputs
 
+    describe "sizeOfCoin" $ do
+        let coinToWord64Clamped = fromMaybe maxBound . Coin.toWord64
+        let cborSizeOfCoin =
+                TxSize
+                . fromIntegral
+                . BS.length
+                . CBOR.toStrictByteString
+                . CBOR.encodeWord64 . coinToWord64Clamped
+
+        let isBoundary c =
+                sizeOfCoin c /= sizeOfCoin (c `Coin.difference` (Coin 1))
+                || sizeOfCoin c /= sizeOfCoin (c `Coin.add` (Coin 1))
+
+        it "matches the size of the Word64 CBOR encoding" $
+            property $ checkCoverage $
+                forAll genEncodingBoundaryLovelace $ \l -> do
+                    let c = fromCardanoLovelace l
+                    let expected = cborSizeOfCoin c
+
+                    -- Use a low coverage requirement of 0.01% just to
+                    -- ensure we see /some/ amount of every size.
+                    let coverSize s = cover 0.01 (s == expected) (show s)
```

Comment:
I checked what would happen with this very low coverage amount, and it does seem to cause QuickCheck to execute a large number of iterations to verify that the coverage is complete.

However, this property executes very quickly, so it seems fine!